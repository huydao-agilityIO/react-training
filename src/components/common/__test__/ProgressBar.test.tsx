import { ProgressBar } from '..';

describe('Button Component', () => {
  it('match snapshot with primary variant', () => {
    const { asFragment } = testLibReactUtils.render(
      <ProgressBar variant="default" size="md" label="Default" />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('match snapshot with success variant', () => {
    const { asFragment } = testLibReactUtils.render(
      <ProgressBar variant="success" size="md" label="Success" />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('match snapshot with danger variant', () => {
    const { asFragment } = testLibReactUtils.render(
      <ProgressBar variant="danger" size="md" label="Danger" />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('match snapshot with warning variant', () => {
    const { asFragment } = testLibReactUtils.render(
      <ProgressBar variant="warning" size="md" label="Warning" />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  it('renders with correct progress value', () => {
    const { getByText } = testLibReactUtils.render(<ProgressBar value={50} />);

    expect(getByText('50%')).toBeInTheDocument();
  });
});
