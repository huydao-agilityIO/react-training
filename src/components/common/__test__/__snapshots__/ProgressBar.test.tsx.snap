// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component match snapshot with danger variant 1`] = `
<DocumentFragment>
  <div
    class="chakra-stack css-nhuyyz"
  >
    <h2
      class="chakra-heading css-1xon16b"
    >
      Danger
    </h2>
    <p
      class="chakra-text css-1cujaad"
    >
      %
    </p>
    <div
      class="css-971dyu"
    >
      <div
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="0"
        class="css-d56dym"
        role="progressbar"
        style="width: 0%;"
      />
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`Button Component match snapshot with primary variant 1`] = `
<DocumentFragment>
  <div
    class="chakra-stack css-nhuyyz"
  >
    <h2
      class="chakra-heading css-1xon16b"
    >
      Default
    </h2>
    <p
      class="chakra-text css-1vmm8s8"
    >
      %
    </p>
    <div
      class="css-971dyu"
    >
      <div
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="0"
        class="css-d737ol"
        role="progressbar"
        style="width: 0%;"
      />
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`Button Component match snapshot with success variant 1`] = `
<DocumentFragment>
  <div
    class="chakra-stack css-nhuyyz"
  >
    <h2
      class="chakra-heading css-1xon16b"
    >
      Success
    </h2>
    <p
      class="chakra-text css-160sqmu"
    >
      %
    </p>
    <div
      class="css-971dyu"
    >
      <div
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="0"
        class="css-1eejryx"
        role="progressbar"
        style="width: 0%;"
      />
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`Button Component match snapshot with warning variant 1`] = `
<DocumentFragment>
  <div
    class="chakra-stack css-nhuyyz"
  >
    <h2
      class="chakra-heading css-1xon16b"
    >
      Warning
    </h2>
    <p
      class="chakra-text css-ne6ooi"
    >
      %
    </p>
    <div
      class="css-971dyu"
    >
      <div
        aria-valuemax="100"
        aria-valuemin="0"
        aria-valuenow="0"
        class="css-obqp8r"
        role="progressbar"
        style="width: 0%;"
      />
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
